src/doc/rustc-dev-guide/src/tests/directives.md

```diff
@@ -262,6 +262,20 @@ Consider writing the test as a proper incremental test instead.
 
 </div>
 
+#### The edition directive
+
+The `//@ edition` directive can take an exact edition, a bounded half-open range of editions or a left-bounded half-open range of editions, this affects which edition is used by `./x test` to run the test. For example:
+
+- A test with the `//@ edition: 2018` directive will only run under the 2018 edition.
+- A test with the `//@ edition: 2015..2021` directive can be run under both the 2015 and 2018 editions. However, CI will only run the test with the loweest edition possible (2015 in this case).
+- A test with the `//@ edition: 2018..` directive will run under any edition greater or equal than 2018. However, CI will only run the test with the lowest edition possible (2018 in this case).
+
+You can also force `./x test` to use a specific edition by passing the `-- --edition=` argument. However, tests with the `//@ edition` directive will only be run if the value of the `--edition` either matches the exact edition or is inside the edition range. For example, if we run `./x edition -- --edition=2015`:
```

Comment:
> directive will only be run if the value of the `--edition` either matches the exact edition or is inside the edition range

Wait, that's not at all accurate. We don't *skip* or *ignore* tests that don't match the provided `--edition`/`--test-args=--edition` but instead the requested edition gets essentially *clamped*.

I know that this feature was requested in the MCP (§ *Add a compiletest flag to only test the default edition*) and I'm all for it but it just doesn't match the current behavior of `--edition` (I do find the current behavior to be highly confusing but that's how it's currently impl'ed if I'm not mistaken).

Reply:
I'm fine with keeping the (odd) clamping logic (`edition_to_test`) for now adding the proposed filter logic in a separate PR (either via new flag or by changing the behavior of `--edition`) but then the docs need to be updated.

Alternatively, I'm also fine with switching to the filtering logic in this PR (in which case the docs would become accurate).